### ISSUE TITLE
Beta 1.6.14 causes connection issues with War Thunder and possibly for all other apps

### ISSUE THREAD

**Issue by Zer0Key** [safing/portmaster#1601]
**What happened**:
War Thunder reports connection issues to game servers and heavy packet loss.
After a failed attempt to join a match with a "connection error", a second attempt will succeed but with heavy packet loss of up to 60% and ping fluctuations of >50ms.
Shutting down portmaster resolves the issue.

**What did you expect to happen?**:
No connection issues or packet loss.

**How did you reproduce it?**:
Launch the game and attempt to play a match.

**Debug Information**:

[War Thunder Portmaster App debug.md](https://github.com/user-attachments/files/16037157/War.Thunder.Portmaster.App.debug.md)  All instances of public facing IP were replaced by 127.0.0.1

[Portmaster Debug.md](https://github.com/user-attachments/files/16037158/Portmaster.Debug.md)

**Other info**:
This issue possibly affects all connections but I don't have the technical knowledge to verify this suspicion. However, I've run into weird delayed and failed connections with multiple websites since switching to the Beta channel. Specifically speed tests will randomly fail with connection errors or will produce wildly varying results with re-tests. All of which disappears if I shut down portmaster.
Could be related to https://github.com/safing/portmaster/issues/1141

**Comment by Zer0Key** [comment]
[Portmaster Debug.md](https://github.com/user-attachments/files/16089353/Portmaster.Debug.md)
[War Thunder Portmaster App debug.md](https://github.com/user-attachments/files/16089354/War.Thunder.Portmaster.App.debug.md)


Re-tested with Beta14, issue persists.

**Comment by Raphty** [comment]
from the debug:

<summary>Unexpected Logs</summary>

```
240703 20:27:55.327 t2/handler:163 > ERRO 110 kext: src/device.rs:159 failed to inject packet: filter-engine: failed to begin transaction: STATUS_FWP_TXN_IN_PROGRESS
240703 20:28:13.329 t2/handler:163 > ERRO 112 kext: src/device.rs:159 failed to inject packet: filter-engine: failed to begin transaction: STATUS_FWP_TXN_IN_PROGRESS
240703 20:28:14.326 t2/handler:163 > ERRO 114 kext: src/device.rs:159 failed to inject packet: filter-engine: failed to begin transaction: STATUS_FWP_TXN_IN_PROGRESS
240703 20:28:26.384 t2/handler:163 > ERRO 116 kext: src/device.rs:159 failed to inject packet: filter-engine: failed to begin transaction: STATUS_FWP_TXN_IN_PROGRESS
240703 20:28:36.471 connection:516 > WARN 118 network: failed to get process and profile of PID 6196: process does not exist
240703 20:30:34.325 t2/handler:163 > ERRO 120 kext: src/device.rs:159 failed to inject packet: filter-engine: failed to begin transaction: STATUS_FWP_TXN_IN_PROGRESS
240703 20:30:36.325 t2/handler:163 > ERRO 122 kext: src/device.rs:159 failed to inject packet: filter-engine: failed to begin transaction: STATUS_FWP_TXN_IN_PROGRESS
240703 20:30:36.325 t2/handler:163 > ERRO 124 kext: src/device.rs:159 failed to inject packet: filter-engine: failed to begin transaction: STATUS_FWP_TXN_IN_PROGRESS
240703 20:32:22.379 ss/process:319 > WARN 126 process: failed to get cmdline (PID 21356): could not get CommandLine: cannot read process PEB
240703 20:32:22.385 ss/process:343 > WARN 128 process: failed to get the process environment (PID 21356): could not get environment variables: cannot read process PEB
240703 20:35:01.526 CURRENT TIME
```

**Comment by Noir16** [comment]
I'm having the same issue with Eternal Return (Unity-based Battle Royale - AOS Game), try to make a multiplayer match in the game and when I try to make a successful match on the server, Portmaster blocks it.

**Comment by Zer0Key** [comment]
Issue persists in 1.6.23
Would love to post the logs but the "copy debug info" button doesn't work.

**Comment by tomasz1986** [comment]
I think that another problematic application may be https://github.com/rustdesk/rustdesk. With the current stable release (v1.6.10), it works flawlessly, however with the beta (tested with v1.6.24 and v1.6.25), there is an error inside RustDesk about being unable to connect to the network. This happens despite allowing the application to connect to the Internet inside Portmaster.

I am sure that the problem is caused by something that has changed in the beta version, because if I downgrade from it to the stable version with the exact same configuration, RustDesk immediately starts working properly again.

Beta v1.6.25 (stuck at connecting):

![image](https://github.com/user-attachments/assets/98f6a5ba-4058-4522-8489-34520b418b4d)

Stable v1.6.10 (connects with no issues):

![image](https://github.com/user-attachments/assets/ab3f5abf-8599-4457-9d0f-bd5ec411440a)

What is strange that inside Portmaster itself, there are no blocked connections reported. Also, I've tried allowing all incoming and outcoming network access, disabling all filter lists, etc., but at the end, Portmaster still blocks the connection somehow.

I can find these unexpected logs in the debug information:

```
2024-11-01 18:23:03.314 ERR t2/handler:163 > 006 kext: src/device.rs:159 failed to inject packet: filter-engine: failed to begin transaction: STATUS_FWP_TXN_IN_PROGRESS
2024-11-01 18:23:05.323 ERR t2/handler:163 > 008 kext: src/device.rs:159 failed to inject packet: filter-engine: failed to begin transaction: STATUS_FWP_TXN_IN_PROGRESS
2024-11-01 18:23:06.313 ERR t2/handler:163 > 010 kext: src/device.rs:159 failed to inject packet: filter-engine: failed to begin transaction: STATUS_FWP_TXN_IN_PROGRESS
2024-11-01 18:23:08.314 ERR t2/handler:163 > 012 kext: src/device.rs:159 failed to inject packet: filter-engine: failed to begin transaction: STATUS_FWP_TXN_IN_PROGRESS
2024-11-01 18:23:09.320 ERR t2/handler:163 > 014 kext: src/device.rs:159 failed to inject packet: filter-engine: failed to begin transaction: STATUS_FWP_TXN_IN_PROGRESS
2024-11-01 18:23:28.319 ERR t2/handler:163 > 016 kext: src/device.rs:159 failed to inject packet: filter-engine: failed to begin transaction: STATUS_FWP_TXN_IN_PROGRESS
```

**Comment by Zer0Key** [comment]
Issue was resolved with 1.6.26
